mathesar/views/api.py

```diff
@@ -21,11 +22,18 @@ def create(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
 
-class TableViewSet(viewsets.ReadOnlyModelViewSet):
+class TableViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin,
+                   CreateModelMixin):
     queryset = Table.objects.all().order_by('-created_at')
     serializer_class = TableSerializer
     pagination_class = DefaultLimitOffsetPagination
 
+    def create(self, request):
+        if "data_file_pk" in request.data:
+            return create_table_from_datafile(request)
+        else:
```

Comment:
I don't think we want to support other types of table creation right now, since it will just create an empty `Table` object in the Django DB that isn't backed by any underlying table in the `mathesar` DB.